tests/tests_service.yml

```diff
@@ -0,0 +1,313 @@
+- name: Test firewall user defined services
+  hosts: all
+  become: true
+  tasks:
+    - name: Test firewall user defined services
+      block:
+
+        # INIT TEST
+
+        - name: Start from clean slate
+          include_role:
+            name: linux-system-roles.firewall
+          vars:
+            firewall:
+              - previous: replaced
+
+        # Verify service states are as intended
+
+        - name: get all services
+          command: firewall-cmd --permanent --get-services
+          register: result
+
+        - name: Fail if http not present after reset
+          fail:
+            msg: Reset default resulted in unexpected state
+          when: result.stdout.find("http ") == -1
```

Comment:
This will fail if the list looks like `ldap ftp http`

I suggest using `firewall-cmd --permanent --query-service=<name>` rather than parsing the stdout of `--get-services` e.g.

```yaml

        - name: Fail if http not present after reset
          command: firewall-cmd --permanent --query-service=http
```

Reply:
I've tried this, it only fetches from the list of services in the default zone. I'll think of a better way, but so far that will not work. Further down I introduced a new test to display the functionality of the command while also using it to test how the user defined services work with zone.

Reply:
Here is a screenshot to illustrate https://drive.google.com/file/d/1OVHSG7l1PdPfmQ21F3QPofqJU-HEGXwI/view?usp=sharing

Reply:
hmm - ok - if you really need to use `--get-services`, then you'll have to use a regex to parse the stdout.

https://docs.ansible.com/ansible/2.9/user_guide/playbooks_tests.html

The pattern has to match something like `(^|\s)theservicename(\s|$)` - beginning of string or space, followed by the name, followed by space or end of string - something like that